uniffi_docs/src/lib.rs

```diff
@@ -2,20 +2,110 @@
 }
 
+impl FromStr for Function {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        // Converting comment into markdown first.
+        let md = s.replace("/// ", "").replace("///", "");
+
+        // Prepare the different buffers for the parse.
+        let mut description_buff = String::new();
+
+        let mut args_values_buff: Vec<String> = Vec::new();
+        let mut args_keys_buff: Vec<String> = Vec::new();
+
+        let mut return_description_buff = String::new();
+
+        // The current parsing stage.
+        let mut stage = ParseStage::DESCRIPTION;
+
+        // A new parser
+        let parser = Parser::new(&md);
+
+        for event in parser {
+            match event {
+                Event::Start(Tag::Heading(H1, _, _)) => match stage {
+                    ParseStage::DESCRIPTION => stage = ParseStage::ARGS,
+                    ParseStage::ARGS => stage = ParseStage::RETURN,
+                    ParseStage::RETURN => {}
+                },
+                Event::Text(s) => match stage {
+                    ParseStage::DESCRIPTION => {
+                        description_buff.push_str(&s);
+                        description_buff.push('\n');
+                    }
+                    ParseStage::ARGS => {
+                        if s.to_lowercase() == "arguments" {
+                            continue;
+                        }
+                        args_values_buff.push(s.to_string());
+                    }
+                    ParseStage::RETURN => {
+                        if s.to_lowercase() == "returns" {
+                            continue;
+                        }
+                        return_description_buff.push_str(&s);
+                        return_description_buff.push('\n');
+                    }
+                },
+                Event::Code(s) => {
+                    args_keys_buff.push(s.to_string());
+                }
+                _ => {}
+            }
+        }
+
+        let mut arguments_descriptions = HashMap::with_capacity(args_keys_buff.len());
+        args_keys_buff
+            .into_iter()
+            .zip(args_values_buff.into_iter())
+            .for_each(|(k, v)| {
+                arguments_descriptions.insert(k, v.replace("-", "").trim().to_string());
+            });
+
+        let return_description = if return_description_buff.is_empty() {
+            None
+        } else {
+            Some(return_description_buff)
+        };
+
+        if arguments_descriptions.is_empty() && return_description.is_none() {
+            return Ok(Function {
+                description: md.to_string(),
+                arguments_descriptions,
+                return_description,
+            });
+        }
+
+        Ok(Function {
+            description: description_buff,
+            arguments_descriptions,
+            return_description,
+        })
+    }
+}
+
+enum ParseStage {
+    DESCRIPTION,
```

Comment:
Enum variants should be named using CamelCase.
See: https://doc.rust-lang.org/rust-by-example/custom_types/enum.html

Reply:
Nice catch. Just passed clippy and warned me about the issue. Fixed at af2117d2b711d7defa34cbb9686baf47fcbc3d25 .